packages/headless/src/features/commerce/facets/facet-set/facet-set-slice.ts

```diff
@@ -113,24 +197,62 @@ function updateStateFromFacetResponse(
   facetRequest.numberOfValues = facetResponse.values.length;
   facetRequest.field = facetResponse.field;
   facetRequest.type = facetResponse.type;
-  facetRequest.values = (facetResponse as FacetResponse).values.map(
-    convertFacetValueToRequest
-  );
+  facetRequest.values = getFacetRequestValuesFromFacetResponse(facetResponse);
   facetRequest.preventAutoSelect = false;
 }
 
+function getFacetRequestValuesFromFacetResponse(
+  facetResponse: AnyFacetResponse
+) {
+  switch (facetResponse.type) {
+    case 'numericalRange':
+      return convertToRangeRequests(
+        (facetResponse as NumericFacetResponse).values
+      );
+    case 'dateRange': // TODO
+    case 'hierarchical': // TODO
+    case 'regular':
+    default:
+      return (facetResponse as RegularFacetResponse).values.map(
```

Comment:
why the cast?

Reply:
Right now it's because I'm defaulting to treating the facet response as a "regular" facet response if its type doesn't match any of the currently handled facetResponse.type values (i.e., numericalRange and regular).

However now that I think of it, this is not a good default behavior.  The method should instead not even try to do a conversion and return nothing when the type is not yet supported.